tests/test_java.py

```diff
@@ -670,31 +672,60 @@ def test_java_different_basename(
         ) as container:
             application_pid = container.attrs["State"]["Pid"]
             profile = snapshot_pid_profile(profiler, application_pid)
-            if change_argv0:
-                # we changed basename - we should have run the profiler
-                assert profile.app_metadata is not None
-                assert (
-                    os.path.basename(profile.app_metadata["execfn"])
-                    == os.path.basename(profile.app_metadata["exe"])
-                    == java_notjava_basename
-                )
-                assert_function_in_collapsed(f"{java_notjava_basename};", profile.stacks)
-                assert_collapsed(profile.stacks)
-            else:
-                # we didn't change basename - we should not have run the profiler due to a different basename.
-                assert profile.stacks == Counter(
-                    {f"{java_notjava_basename};[Profiling skipped: profiling this JVM is not supported]": 1}
-                )
-                log_records = list(
-                    filter(
-                        lambda r: r.message
-                        == "Non-java basenamed process (cannot get Java version), skipping... (disable"
-                        " --java-safemode=java-extended-version-checks to profile it anyway)",
-                        caplog.records,
-                    )
-                )
-                assert len(log_records) == 1
-                assert os.path.basename(log_record_extra(log_records[0])["exe"]) == java_notjava_basename
+            assert profile.app_metadata is not None
```

Comment:
You merged the 2 cases but technically the 3rd case (that previously triggered the "non-java basename") is now handled via libjvm. Right?